src/pages/StockPage.js

```diff
@@ -22,40 +23,22 @@ const DATA_TYPES_SYNCHRONISED = ['Item', 'ItemBatch', 'ItemCategory'];
 * @prop   {func}                navigateTo  CallBack for navigation stack.
 * @state  {Realm.Results}       items       Contains all Items stored on the local database.
 */
-export class StockPage extends GenericPage {
+export class StockPage extends React.Component {
   constructor(props) {
     super(props);
-    this.state.sortBy = 'name';
-    this.state.items = props.database.objects('Item');
-    this.state.columns = [
-      {
-        key: 'code',
-        width: 1,
-        title: tableStrings.item_code,
-        sortable: true,
-      },
-      {
-        key: 'name',
-        width: 5,
-        title: tableStrings.item_name,
-        sortable: true,
-      },
-      {
-        key: 'totalQuantity',
-        width: 1,
-        title: tableStrings.stock_on_hand,
-        sortable: true,
-        alignText: 'right',
-      },
-    ];
-    this.dataTypesSynchronised = DATA_TYPES_SYNCHRONISED;
-    this.getFilteredSortedData = this.getFilteredSortedData.bind(this);
+    this.state = {
+      items: props.database.objects('Item'),
+    };
+    autobind(this);
   }
 
   /**
    * Returns updated data according to searchTerm, sortBy and isAscending.
    */
-  getFilteredSortedData(searchTerm, sortBy, isAscending) {
+  refreshData(searchTerm = this.searchTerm, sortBy = this.sortBy, isAscending = this.isAscending) {
```

Comment:
All the defaults don't appear to be actually declared in the constructor anyway?